src/backup_observer.py

```diff
@@ -168,6 +169,16 @@ def _on_restore_backup_action(self, event: ActionEvent) -> None:
             event.fail("Wrong S3 configuration on restore backup action. Check S3 integration.")
             return
 
+        try:
+            s3_client = backup.S3Client(s3_parameters)
+            if not s3_client.exists_backup(backup_id):
+                event.fail(f"backup-id {backup_id} does not exist")
```

Comment:
I'm wondering if we should fail here or exit with success and still print this output so that the action is idempotent